### DISCUSSION TITLE
client.load_data_type_definitions does not create expected class for custom structure in `ua` module

### DISCUSSION THREAD

**Discussion by scotthz** [FreeOpcUa/opcua-asyncio#1419] [Q&A]
I am trying to instantiate a custom structure type following the guidance in https://github.com/FreeOpcUa/opcua-asyncio/discussions/839 (Usage of Extension Objects). The structure (ScanSettings) is defined and browsable. But after calling `load_data_type_definitions` I get an error trying to instantiate the expected class. Running the following snippet prints the structure type's browse name then hits an `AttributeError` on `ua.ScanSettings()`.

What am I missing here?

```python
        root = client.get_root_node()
        scan_settings_node = await root.get_child(["0:Types", "0:DataTypes", "0:BaseDataType", "0:Structure", "3:ScanSettings"])
        print(await scan_settings_node.read_browse_name())

        await client.load_data_type_definitions()
        scan_settings = ua.ScanSettings()
```

```text
Connecting to opc.tcp://192.168.37.205:4840
QualifiedName(NamespaceIndex=3, Name='ScanSettings')
...
    scan_settings = ua.ScanSettings()
                    ^^^^^^^^^^^^^^^
AttributeError: module 'asyncua.ua' has no attribute 'ScanSettings'
```

**Comment by schroeder-** [comment]
Maybe the server is doesn't support the new datatype description. 
Try to load `load_type_definitions` instead of `load_data_type_definitions`.
If this doesn't work can you activate logging and post the output?

**Reply by scotthz** [reply]
Thank you for your quick response! Your suggestion to use `load_type_definitions` is working and enables me to move forward.
I do get this error message, I guess it is safe to ignore for now?

`Deprecated since spec 1.04, call load_data_type_definitions`

**Reply by schroeder-** [reply]
You can ignore that message, for your usecase. Because your server doesn't support the 1.04 datatype defintions.

**Comment by m-stredansky** [comment]
If you have custom data types on OPC Server,  It would be nice to load them first either by creating task for that or wait for load.

  task_load_types = asyncio.create_task(client.load_type_definitions())
  type_definitions_loaded = await asyncio.wait_for(task_load_types, timeout=10) # Max timeout 10
  
  Method **load_type_definitions** is supported only in versions <= 1.04 and is now deprecated.
  
  For newer versions for loading custom types (structures/extension objects) definition from server use method **load_data_type_definitions**. I recommend either installing version 1.04 and keeping the compatibility or going to a higher version with a new call.
